zaza/openstack/charm_tests/manila/tests.py

```diff
@@ -313,3 +352,17 @@ def test_manila_share(self):
             fip_2, ssh_user_name, privkey, share_path)
         self._validate_testing_file_from_instance(
             fip_2, ssh_user_name, privkey)
+
+        # Restart the share provider
+        if self._restart_share_instance():
+            # Write a testing file on instance #1
+            self._mount_share_on_instance(
+                fip_1, ssh_user_name, privkey, share_path)
+            self._write_testing_file_on_instance(
```

Comment:
Should it be a different testing file, i.e. with a different name?

EDIT: ah no, but you forgot to make use of `_clear_testing_file_on_instance()`?